70_FLIPFluids/app_resources/gridUtils.hlsl

```diff
@@ -1,32 +1,37 @@
 #ifndef _FLIP_EXAMPLE_GRID_UTILS_HLSL
 #define _FLIP_EXAMPLE_GRID_UTILS_HLSL
 
-#ifdef __HLSL_VERSION
 struct SGridData
 {
-    float gridCellSize;
-    float gridInvCellSize;
-    float pad0[2];
+    float32_t gridCellSize;
+    float32_t gridInvCellSize;
 
-    int4 particleInitMin;
-    int4 particleInitMax;
-    int4 particleInitSize;
+    int32_t4 particleInitMin;
+    int32_t4 particleInitMax;
+    int32_t4 particleInitSize;
 
-    float4 worldMin;
-    float4 worldMax;
-    int4 gridSize;
+    float32_t4 worldMin;
+    float32_t4 worldMax;
+    int32_t4 gridSize;
```

Comment:
why are you using 4D vectors for 3D quantities?

Reply:
this is what https://github.com/Devsh-Graphics-Programming/Nabla-Examples-and-Tests/pull/145#discussion_r1796559970 was about